homeassistant/components/starlink/coordinator.py

```diff
@@ -27,12 +44,12 @@ def __init__(self, hass: HomeAssistant, name: str, url: str) -> None:
             update_interval=timedelta(seconds=5),
         )
 
-    async def _async_update_data(self) -> StatusDict:
+    async def _async_update_data(self) -> StarlinkData:
         async with async_timeout.timeout(4):
             try:
                 status = await self.hass.async_add_executor_job(
                     status_data, self.channel_context
                 )
-                return status[0]
+                return StarlinkData(status[0], status[1], status[2])
```

Comment:
```suggestion
                return StarlinkData(*status)
```

Untested but probably works (will test it when I can factory reset if you don't get a chance to before)

Also would be nicer if the lib provided the data in this format instead of here.

Reply:
I will raise an issue with the library, is it fine to keep it here for now?

Reply:
Yes